### PR TITLE
Do not record s3 http request time when it is not available

### PR DESCRIPTION
**Pull request by ywangd**
The metrics of HTTP request time can be unavailable (null) when the request fails on the client side. This PR makes sure we do not attempt to record it when it happens to avoid NPE.

### CODE REVIEW

File: modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java [hunk 1]
```diff
@@ -185,7 +185,27 @@ public final void collectMetrics(Request<?> request, Response<?> response) {
...
+         * Used for APM style metrics to measure statics about performance. This is not for billing.
+         */
+        private void maybeRecordHttpRequestTime(Request<?> request) {
+            final List<TimingInfo> requestTimesIncludingRetries = request.getAWSRequestMetrics()
+                .getTimingInfo()
+                .getAllSubMeasurements(AWSRequestMetrics.Field.HttpRequestTime.name());
+            // It can be null if the request did not reach the server for some reason
+            if (requestTimesIncludingRetries == null) {
+                return;
+            }
```
Comment (ywangd) [inline modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java:201]: This is the main change that prevents a NPE being thrown when it gets iterated below. It's pretty straightforward. But let me know if you think a test is necessary.

**Comment by ywangd** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
Gentle reminder for review. Thanks!

File: modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java [hunk 2]
```diff
@@ -185,7 +185,27 @@ public final void collectMetrics(Request<?> request, Response<?> response) {
...
+
+            final long totalTimeInMicros = getTotalTimeInMicros(requestTimesIncludingRetries);
+            if (totalTimeInMicros == 0) {
+                logger.warn("Expected HttpRequestTime to be tracked for request [{}] but found no count.", request);
```
Comment (idegtiarenko) [inline modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java:205]: Has it ever happened?

**Review by idegtiarenko — APPROVED** [review]

File: modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java (hunk 2 above)
Comment (ywangd) [reply modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java:205]: It happend for me on a local cluster where I forgot to configure the s3 client credentials. The request failed locally and did not have any request time info.

**Pull request merged**